stateproof/db.go

```diff
@@ -86,14 +132,28 @@ func getPendingSigs(tx *sql.Tx) (map[basics.Round][]pendingSig, error) {
 	return rowsToPendingSigs(rows)
 }
 
-func getPendingSigsFromThisNode(tx *sql.Tx) (map[basics.Round][]pendingSig, error) {
-	rows, err := tx.Query("SELECT sprnd, signer, sig, from_this_node FROM sigs WHERE from_this_node=1")
+func getPendingSigsForRound(tx *sql.Tx, rnd basics.Round) ([]pendingSig, error) {
+	rows, err := tx.Query("SELECT sprnd, signer, sig, from_this_node FROM sigs WHERE sprnd=?", rnd)
 	if err != nil {
 		return nil, err
 	}
 	defer rows.Close()
+	tmpmap, err := rowsToPendingSigs(rows)
+	if err != nil {
+		return nil, err
+	}
+	return tmpmap[rnd], nil
+}
 
-	return rowsToPendingSigs(rows)
+func sigExistsInDB(tx *sql.Tx, rnd basics.Round, account Address) (bool, error) {
+	row := tx.QueryRow("SELECT EXISTS ( SELECT 1 FROM sigs WHERE signer=? AND sprnd=?)", account[:], rnd)
```

Comment:
The execution plan for EXISTS is indeed longer, but the return value from this query is a boolean, instead of merklesignature.Signature, which consists of:
- Falcon signature
- Merkle proof (using subset-sum hash)
- Falcon verifier
and a few other small fields, which sums up to about 4288 bytes.

So, even though we SQL query might take a bit longer we avoid `Scan` of a 4KB value.
That seems more performant to me, what do you think?

Reply:
todo: add a comment about perf